pkg/helpers/helpers.go

```diff
@@ -203,3 +208,13 @@ func GetCloudTargetEnv(location string) string {
 		return "AzurePublicCloud"
 	}
 }
+
+//AssignStringIfNotEmpty set the *dst to src if src is not empty string
+func AssignStringIfNotEmpty(dst *string, src string) {
+	if dst == nil {
```

Comment:
Question: Is there an assumption that a nil field will never have defaults populated to it?

Reply:
Yes, in current situation.
All the fields in AzureCloudSpecConfig are not pointer value, so they will never be true.

I add this check to avoid panic if the nil is passed unexpectedly. @tariq1890

Reply:
Yes, in current situation.
All the fields in AzureCloudSpecConfig are not pointer value, so they will never be true.

I add this check to avoid panic if the nil is passed unexpectedly. @tariq1890

Reply:
The title `AssignStringIfNotEmpty` is a bit confusing. The method basically assigns a default string value to a non-nil field that has an Empty string as value.